students/roviram/Lesson03/mail_room.py

```diff
@@ -0,0 +1,114 @@
+"""
+
+donors = [
+        # Each Row represents 1 element
+        ("Miguel Rovira", [100, 200, 300, 400, 500]),
+        ("Emma DaFoe", [200, 40520, 5000, 100, 400]),
+        ("Pepper", [5, 20, 100]),
+        ("Vicky Gonzalez", [600, 9000, 10]),
+        ("Cristin Rovira", [25, 50, 75]),
+    ]
+
+report_list = []
+new_donor = []
+
+# Processing
+def gen_stats(donors):
+    """Generating stats of the donors"""
+    gen_list = []
+    final_list = donors
+    for donor in donors:
+        donations = donor[1]
+        total = sum(donations)
+        number_of_donations = len(donations)
+        avg = round(total / number_of_donations, 2)
+        stats = [donor[0], total, number_of_donations, avg]
+        gen_list.append(stats)
+    return gen_list
+
+def create_report(report_donors):
+    """Creating a report of the mail room donors"""
+    stats_list = gen_stats(report_donors)
+    stats_list.sort(key=lambda stats_list: stats_list[1], reverse=True)
+    print("{:<20} | {:^15} | {:>15} | {:>15}".format("Name of Donor", "Total Donation Amount", "Number of Donations", "Average Gift Amount"))
+    print("-"*89)
+    for index, donor in enumerate(stats_list):
+        print(str(index + 1) + ".", "{:<20}${:^15}{:>15}\t\t\t\t{}{:>15}".format(donor[0], donor[1], donor[2],"$",donor[3]))
+    return report_list
+
+#create_report(donors)
+def intro():
+    """Welcoming the user to the application"""
+    # Intro to the application
+    print("Hello! Welcome to the mailroom! We are excited you are apart of our team.\n")
+
+def thank_you():
+    """Thanking the donor for their donor and adding a new donor if they are new"""
+    reset()
+    while True:
+        list_answer = input("\nThank you notes are always appreciated!\n"
+                            "Want to see the list of donors to send a 'Thank You' too? (type 'Yes' or 'No')")
+        # Cleaning up the Users Answer
+        if list_answer.upper().strip() == 'YES':
+            print(donors)
+
+        # Prompt Donor Name
+        donor_name = input("Please enter the donor name: ")
+        #reset()
+        print("You typed:", donor_name)
+        match = False
+        for donor in donors:
+            if donor_name.lower() == donor[0].lower():
+                print("Cool, I found that donor")
+                match = True
+
+        if not match:
+            print("I didn't find that donor, I am adding them to our database now")
+            donation_amount = int(input("How much did they donate?:"))
+            donors.append((donor_name, [donation_amount]))
+
+        print()
+        print("Time to send a thank you email!")
+        print("Dear", donor_name + ",\n\n"
+        "Thank you for your generous donation of", str(donation_amount) + ".", "\n\nEmail has been sent!")
+
+        menu_answer = input("Are you done? ('Yes' or 'No')")
+        if menu_answer.upper().strip() == 'YES':
+            break
+
+def main_menu():
```

Comment:
Overall this looks pretty good! Just make sure to remove the reset calls (or implement the function).